spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/types/ColumnType.scala

```diff
@@ -84,7 +92,8 @@ object ColumnType {
 
   def fromDriverType(dataType: DataType): ColumnType[_] = {
     val typeArgs = dataType.getTypeArguments.map(fromDriverType)
-    (dataType, dataType.getName) match {
+
+    val standardFromDriverRow: PartialFunction[(DataType, DataType.Name), ColumnType[_]] = {
```

Comment:
Perhaps it would be better to predefine the partial function which takes what it takes now + type args. You wouldn't have create and chain them every time `fromDriverType` is invoked. I know it almost does not affect performance, so I don't insist.